### PR TITLE
PHOENIX-3710 Cannot use lowername data table name with indextool

### PR DESCRIPTION
**Pull request by stoty**
Co-authored-by: Ankit Singhal <ankitsinghal59@gmail.com>
Co-authored-by: Josh Elser <elserj@apache.org>

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  17m  7s |  master passed  |
| +1 :green_heart: |  compile  |   1m 22s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  7s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  7s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 31s |  phoenix-core in master has 972 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  12m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 34s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 34s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 30s |  phoenix-core: The patch generated 34 new + 1271 unchanged - 26 fixed = 1305 total (was 1297)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  javadoc  |   1m 22s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   5m 31s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 217m 51s |  phoenix-core in the patch failed.  |
| -1 :x: |  asflicense  |   0m 31s |  The patch generated 1 ASF License warnings.  |
|  |   | 270m 51s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.IndexExtendedIT |
|   | phoenix.end2end.IndexToolIT |
|   | phoenix.end2end.ClientHashAggregateIT |
|   | phoenix.end2end.index.IndexMetadataIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1071 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 2a8bffe4312a 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 5ef5faa |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/artifact/yetus-general-check/output/whitespace-eol.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 5827 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 32s |  master passed  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 52s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 46s |  master passed  |
| +0 :ok: |  spotbugs  |   3m  0s |  phoenix-core in master has 972 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 46s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 56s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 49s |  phoenix-core: The patch generated 41 new + 1267 unchanged - 30 fixed = 1308 total (was 1297)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 10s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 118m 38s |  phoenix-core in the patch failed.  |
| -1 :x: |  asflicense  |   0m 28s |  The patch generated 1 ASF License warnings.  |
|  |   | 156m  1s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.UpsertSelectIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1071 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 074f651feee9 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 5ef5faa |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/artifact/yetus-general-check/output/whitespace-eol.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 6751 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  17m 18s |  master passed  |
| +1 :green_heart: |  compile  |   1m 16s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  0s |  master passed  |
| +0 :ok: |  spotbugs  |   4m  3s |  phoenix-core in master has 972 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  11m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 16s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m  6s |  phoenix-core: The patch generated 39 new + 1267 unchanged - 30 fixed = 1306 total (was 1297)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   1m  3s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   4m 24s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 116m 46s |  phoenix-core in the patch failed.  |
| -1 :x: |  asflicense  |   0m 31s |  The patch generated 1 ASF License warnings.  |
|  |   | 164m 20s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.index.IndexMetadataIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1071 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 3962b081b052 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 132aed1 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 6579 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by joshelser — APPROVED** [review]
+1

Embarrassing, this one. Twice I said I would come back and "finish" it. Thank you for actually following-through, Istvan.

Noticed SingleCellIndexIT.java is also missing a license header. Let me see if anyone else has a patch and then fix. Test failure seems unrelated to your changes.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java [hunk 1]
```diff
@@ -821,6 +942,7 @@ public static IndexTool runIndexTool(Configuration conf, boolean directApi, bool
                 indexTableName, tenantId, verifyType, disableLoggingType);
         List<String> cmdArgList = new ArrayList<>(Arrays.asList(cmdArgs));
         cmdArgList.addAll(Arrays.asList(additionalArgs));
+        LOGGER.info("Running IndexTool with {}", Arrays.toString(cmdArgList.toArray()), new Exception("Stack Trace"));
```
Comment (joshelser) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java:945]: You want to keep this here?

**Comment by stoty** [comment]
> Noticed SingleCellIndexIT.java is also missing a license header. Let me see if anyone else has a patch and then fix. Test failure seems unrelated to your changes.

It's fixed on master already.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java (hunk 1 above)
Comment (stoty) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java:945]: I found it really useful, I think it's OK for a test case.
I'm going to fix up the exception message to something more descriptive.

Comment (joshelser) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java]: Ok! If you find it useful, then that's great.

**Review by joshelser — APPROVED** [review]

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 22s |  master passed  |
| +1 :green_heart: |  compile  |   0m 52s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 59s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 54s |  phoenix-core in master has 956 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 57s |  phoenix-core: The patch generated 39 new + 1267 unchanged - 30 fixed = 1306 total (was 1297)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  7s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 108m 55s |  phoenix-core in the patch failed.  |
| -1 :x: |  asflicense  |   0m 29s |  The patch generated 1 ASF License warnings.  |
|  |   | 145m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1071 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 4c4492faaf2c 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / edbd6e8 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 6687 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1071/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**